std_execution.bs

```diff
@@ -4128,18 +4128,16 @@ from the operation before the operation completes.
 
         * <i>count</i> is `0` and <i>state</i> is <i>finishing</i>, in which case `pop_front()` returns `nullptr`; or
 
-        * <i>count</i> is greater than `0`, in which case `pop_front()` removes an item from the front of the queue, decrements <i>count</i> by `1`, and returns the item.
-
-    2. <i>Synchronization:</i> This function is thread-safe with respect to `push_back()` and `finish()`. Removing a queue item and decrementing <i>count</i> are executed atomically.
+        * <i>count</i> is greater than `0`, in which case an item is atomically removed from the front of the queue, <i>count</i> is atomically decremented by `1`, and the removed item is returned.
```

Comment:
Can another thread observe the queue state after the item has been removed but before the count is decremented?

Reply:
Addressed by moving to barrier style wording (https://eel.is/c++draft/thread.barrier#class-4), as per Olivier's suggestion.